docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
```

Comment:
@smarterclayton has a good point about facilitating the use of versioned config types. We're working on converting our components to that approach, also (#12245). What would we need beyond a kubectl command to bundle a config file into a ConfigData resource, such as that discussed for secret (#4822)?